kbl2vec/src/main/java/com/foursoft/harness/kbl2vec/transform/topology/placements/AbstractSegmentLocationTransformer.java

```diff
@@ -0,0 +1,99 @@
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ * =========================LICENSE_END==================================
+ */
+package com.foursoft.harness.kbl2vec.transform.topology.placements;
+
+import com.foursoft.harness.kbl.v25.KblNumericalValue;
+import com.foursoft.harness.kbl.v25.KblProtectionArea;
+import com.foursoft.harness.kbl.v25.KblUnit;
+import com.foursoft.harness.kbl2vec.core.Query;
+import com.foursoft.harness.kbl2vec.core.TransformationContext;
+import com.foursoft.harness.kbl2vec.core.TransformationResult;
+import com.foursoft.harness.kbl2vec.core.Transformer;
+import com.foursoft.harness.vec.v2x.VecNumericalValue;
+import com.foursoft.harness.vec.v2x.VecSegmentLocation;
+import com.foursoft.harness.vec.v2x.VecTopologySegment;
+import com.foursoft.harness.vec.v2x.VecUnit;
+
+public abstract class AbstractSegmentLocationTransformer<S> implements Transformer<S, VecSegmentLocation> {
+
+    @Override
+    public TransformationResult<VecSegmentLocation> transform(final TransformationContext context, final S source) {
+        final VecSegmentLocation destination = new VecSegmentLocation();
+
+        final LocationData locationData = extractLocationData(source);
+        destination.setIdentification(locationData.identification);
+
+        final TransformationResult.Builder<VecSegmentLocation> builder = TransformationResult.from(destination);
+
+        if (source instanceof final KblProtectionArea protectionArea) {
+            builder.withLinker(Query.of(protectionArea::getParentSegment), VecTopologySegment.class,
+                               VecSegmentLocation::setReferencedSegment);
+
+            if (locationData.absoluteLocation != null) {
+                return builder
+                        .withDownstream(KblNumericalValue.class, VecNumericalValue.class,
+                                        Query.of(locationData.absoluteLocation),
+                                        VecSegmentLocation::setOffset)
+                        .build();
+            }
+
+            destination.setOffset(calculateAbsoluteLocation(source, locationData));
+            return builder
+                    .withLinker(Query.of(locationData::baseUnit), VecUnit.class,
+                                (dest, unit) -> dest.getOffset().setUnitComponent(unit))
+                    .build();
+        }
+        return TransformationResult.noResult();
+    }
+
+    private VecNumericalValue calculateAbsoluteLocation(final S source, final LocationData locationData) {
+        final VecNumericalValue absoluteLocation = new VecNumericalValue();
+
+        if (Double.isNaN(locationData.relativeLocation)) {
```

Comment:
Good think to catch the error case. I am wondering what is the best result that should be returned, because that will result in schema validation errors (value & unit are mandatory for a numericalvalue). I think, if relative is NaN then absolute should be NaN as well.